### PR TITLE
code review changes

### PR DESCRIPTION
**Pull request by ZappoMan**
this PR has no material effect on behavior, it was only small code review changes (mostly renaming variables) requested for PR #9862 that accidentally got merged before the changes went in.

- use of the word "frame" has been replaced with "block"
- the fade in on noise gate is only over 1/10th to avoid possible lose of transients opening the gate
- the fade calculation is optimized by moving the divide out of the loop
- two if's that are mutually exclusive are turned into an `else if` to optimize performance.

### CODE REVIEW

**Comment by ZappoMan** [comment]
#TEST PLAN

## NOISE-GATE-01 Test
Objective: Verify that noise gate opens and closes as expected.
Precondition: Enter any domain, but preferably one with limited audio, such that you are the only person in the domain. Make sure the Audio Noise Reduction is enabled.
Step 1: Run scripts/system/audioScope.js
Step 2: Enable the audio scope, and enable "auto pause".
Step 3: Try being quite and speaking alternately, verify that the audio scope visualization automatically pauses when you are mostly quite, and starts when you make noise.

## NOISE-GATE-02 Test
Objective: Verify that when the noise gate closes and your client sends silent packets that there is no distortion in the echoed audio.
Precondition: Enter any domain, make sure that the domain is totally silent other than your own client. Make sure the Audio Noise Reduction is enabled.
Step 1: Run scripts/system/audioScope.js
Step 2: Enable the audio scope, and enable "auto pause".
Step 3: Enable Server Loopback
Step 4: Try being quite and speaking alternately, verify that the audio scope visualization automatically pauses when you are mostly quite, and starts when you make noise.
Step 5: Verify that after being quiet and beginning to speak, that the echoed audio is a close match to the outbound audio wave form. It should not contain significant distortions. You should listen closely for things like apparent "type writer" sounds or other distortions, and you should inspect the paused audio scope wave form to make sure it looks like a match.

## NOISE-GATE-03 Test
Objective: Verify that reports of noise gate opening and closing are always in sequential pairs, that we don't report that the noise gate closes more than once in a row.
Precondition: Enter any domain, make sure that the domain is totally silent other than your own client. Make sure the Audio Noise Reduction is enabled.
Step 1: Run interface.
Step 2: Open Edit > Console.
Step 3: Enter the following commands:
Audio.noiseGateOpened.connect(function() { print("opened"); }); Audio.noiseGateClosed.connect(function() { print("closed"); });
Step 4: Verify that in the log you see "opened" followed by "closed" and then repeated each time the noise gate opens or closes.

File: libraries/audio-client/src/AudioNoiseGate.cpp [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
-    _didClipInLastFrame(false),
+    _quietestBlock(std::numeric_limits<float>::max()),
+    _loudestBlock(0.0f),
+    _didClipInLastBlock(false),
     _dcOffset(0.0f),
     _measuredFloor(0.0f),
     _sampleCounter(0),
     _isOpen(false),
-    _framesToClose(0)
+    _buffersToClose(0)
```
Comment (kencooke) [inline libraries/audio-client/src/AudioNoiseGate.cpp]: "buffers" -> "blocks" ?

File: libraries/audio-client/src/AudioNoiseGate.cpp [hunk 2]
```diff
@@ -37,7 +37,7 @@ void AudioNoiseGate::removeDCOffset(int16_t* samples, int numSamples) {
     //
     //  DC Offset correction
     //
-    //  Measure the DC offset over a trailing number of frames, and remove it from the input signal.
+    //  Measure the DC offset over a trailing number of buffers, and remove it from the input signal.
```
Comment (kencooke) [inline libraries/audio-client/src/AudioNoiseGate.cpp]: "buffers" -> "blocks" ?

File: libraries/audio-client/src/AudioNoiseGate.cpp [hunk 3]
```diff
@@ -51,7 +51,7 @@ void AudioNoiseGate::removeDCOffset(int16_t* samples, int numSamples) {
     // Update measured DC offset
     measuredDcOffset /= numSamples;
     if (_dcOffset == 0.0f) {
-        // On first frame, copy over measured offset
+        // On first buffer, copy over measured offset
```
Comment (kencooke) [inline libraries/audio-client/src/AudioNoiseGate.cpp]: "buffers" -> "blocks" ?

File: libraries/audio-client/src/AudioNoiseGate.cpp [hunk 4]
```diff
@@ -69,30 +69,30 @@ void AudioNoiseGate::gateSamples(int16_t* samples, int numSamples) {
     //
     //  NOISE_GATE_HEIGHT:  How loud you have to speak relative to noise background to open the gate.
     //                      Make this value lower for more sensitivity and less rejection of noise.
-    //  NOISE_GATE_WIDTH:   The number of samples in an audio frame for which the height must be exceeded
+    //  NOISE_GATE_WIDTH:   The number of samples in an audio buffer for which the height must be exceeded
```
Comment (kencooke) [inline libraries/audio-client/src/AudioNoiseGate.cpp]: "buffers" -> "blocks" ?

File: libraries/audio-client/src/AudioNoiseGate.cpp [hunk 5]
```diff
@@ -69,30 +69,30 @@ void AudioNoiseGate::gateSamples(int16_t* samples, int numSamples) {
...
     //                      to open the gate.
-    //  NOISE_GATE_CLOSE_FRAME_DELAY:  Once the noise is below the gate height for the frame, how many frames
+    //  NOISE_GATE_CLOSE_BLOCK_DELAY:  Once the noise is below the gate height for the buffer, how many buffers
     //                      will we wait before closing the gate.
-    //  NOISE_GATE_FRAMES_TO_AVERAGE:  How many audio frames should we average together to compute noise floor.
+    //  NOISE_GATE_BLOCKSS_TO_AVERAGE:  How many audio buffers should we average together to compute noise floor.
```
Comment (kencooke) [inline libraries/audio-client/src/AudioNoiseGate.cpp]: "buffers" -> "blocks" ?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 956fdd09339378ae25ecc17a3fbb262d3f0c59ad available at: https://gist.github.com/23daee414e38c0175298c6830f7bdde7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9872-956fdd0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9872-956fdd0.dmg

**Comment by OmegaHeron** [comment]
Windows 10, QA Pass

NOISE-GATE-01 Test *Pass*
NOISE-GATE-03 Test *Pass*

**Pull request merged**